### PR TITLE
Export validation

### PR DESCRIPTION
**Pull request by knockknockyoo**
1. archive / upload s3 bucket
2. constants(s3 bucket purpose)
- release, archive_release
3. create a file on the buffer memory to upload s3 bucket

### CODE REVIEW

File: src/metadata_export.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        props = data_record.get(PROPERTIES)
+        buf.write('\t'.join(map(str, props.keys())).encode() + b'\n')
+        buf.write('\t'.join(map(str, props.values())).encode() + b'\n')
+        buf.seek(0)
+        buf.name = f"{file_name}.{file_type}"
+        return buf
+
+
+# Private class
+class ExportValidator:
```
Comment (vshand11) [inline src/metadata_export.py:112]: Class name should be ExportMetadata here.

File: src/metadata_export.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    def __init__(self, mongo_dao, submission, s3_service):
+        self.log = get_logger(TYPE_EXPORT_METADATA)
+        self.mongo_dao = mongo_dao
+        self.submission = submission
+        self.s3_service = s3_service
+
+    def validate(self):
```
Comment (vshand11) [inline src/metadata_export.py:119]: function name should be export_data_to_file?

File: src/metadata_export.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        submission_id, bucket_name = self.get_submission_info()
+        records = self.mongo_dao.get_dataRecords(submission_id, None)
+        # create file data and file name
+        files_to_export = []
+        for r in records:
```
Comment (vshand11) [inline src/metadata_export.py:124]: Need group dataRecords by nodeType and export all nodes in a group to a file.

**Review by vshand11 — CHANGES_REQUESTED** [review]
Please read comments and make changes.

File: src/metadata_export.py (hunk 1 above)
Comment (knockknockyoo) [reply src/metadata_export.py]: updated!

File: src/metadata_export.py (hunk 2 above)
Comment (knockknockyoo) [reply src/metadata_export.py]: updated!

File: src/metadata_export.py (hunk 3 above)
Comment (knockknockyoo) [reply src/metadata_export.py]: updated!

**Comment by knockknockyoo** [comment]
1. Line 12, remove the line 12 and put it in s3service _init. To make the client fresh every time a S3Service is created.
 - **updated**
2. Line 53, insert statement, export_validator = None to release memory
- **updated**
3. class ValidationFile is not needed. Just add create_file as a static function within the class, ExportMetaData, since it is not called by other class.
- **removed**

**Review by vshand11 — APPROVED** [review]
The latest version passed review and local testing.

**Pull request merged**